src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlTypes/SqlFileStream.Windows.cs

```diff
@@ -0,0 +1,959 @@
+            }
+
+            #if NETFRAMEWORK
+            // Ensure the running code has permission to read/write the file
+            DemandAccessPermission(path, access);
+            #endif
+
+            Interop.NtDll.CreateOptions createOptions = 0;
+            Interop.NtDll.CreateDisposition createDisposition = 0;
+            Interop.NtDll.DesiredAccess desiredAccess = Interop.NtDll.DesiredAccess.FILE_READ_ATTRIBUTES |
+                                                        Interop.NtDll.DesiredAccess.SYNCHRONIZE;
+            FileShare shareAccess = 0;
+
+            switch (access)
+            {
+                case FileAccess.Read:
+                    desiredAccess |= Interop.NtDll.DesiredAccess.FILE_READ_DATA;
+                    shareAccess = FileShare.Delete | FileShare.ReadWrite;
+                    createDisposition = Interop.NtDll.CreateDisposition.FILE_OPEN;
+                    break;
+
+                case FileAccess.Write:
+                    desiredAccess |= Interop.NtDll.DesiredAccess.FILE_WRITE_DATA;
+                    shareAccess = FileShare.Delete | FileShare.Read;
+                    createDisposition = Interop.NtDll.CreateDisposition.FILE_OVERWRITE;
+                    break;
+
+                case FileAccess.ReadWrite:
+                    desiredAccess |= Interop.NtDll.DesiredAccess.FILE_READ_DATA |
+                                      Interop.NtDll.DesiredAccess.FILE_WRITE_DATA;
+                    shareAccess = FileShare.Delete | FileShare.Read;
+                    createDisposition = Interop.NtDll.CreateDisposition.FILE_OVERWRITE;
+                    break;
+
+                // Note: default case is heuristically unreachable due to check above.
+            }
+
+            if ((options & FileOptions.WriteThrough) != 0)
+            {
+                createOptions |= Interop.NtDll.CreateOptions.FILE_WRITE_THROUGH;
+            }
+
+            if ((options & FileOptions.Asynchronous) == 0)
+            {
+                createOptions |= Interop.NtDll.CreateOptions.FILE_SYNCHRONOUS_IO_NONALERT;
+            }
+
+            if ((options & FileOptions.SequentialScan) != 0)
+            {
+                createOptions |= Interop.NtDll.CreateOptions.FILE_SEQUENTIAL_ONLY;
+            }
+
+            if ((options & FileOptions.RandomAccess) != 0)
+            {
+                createOptions |= Interop.NtDll.CreateOptions.FILE_RANDOM_ACCESS;
+            }
+
+            SafeFileHandle fileHandle = null;
+            try
+            {
+                // NOTE: the Name property is intended to reveal the publicly available moniker for the
+                //   FILESTREAM attributed column data. We will not surface the internal processing that
+                //   takes place to create the mappedPath.
+                string mappedPath = InitializeNtPath(path);
+
+                Interop.Kernel32.SetThreadErrorMode(Interop.Kernel32.SEM_FAILCRITICALERRORS, out uint oldMode);
```

Comment:
The OS version being checked was 6.1.7600 - Windows 7 and Windows Server 2008 R2. The only supported case of the check is `true`, so removing it doesn't impact supported OSes.

Reply:
Personally I don't think it's a problem since Windows 7 and Windows Server 2008 R2 have been out of support for 4 years. But if it's an issue, perhaps someone else can add to the discussion?